### PR TITLE
Use new no error sections option

### PR DESCRIPTION
**Pull request by vinistock**
Use the new option introduced in https://github.com/sorbet/sorbet/pull/4078 for Spoom bump.

I locked the version of Sorbet to be greater or equal to the one where the new option is introduced.

### CODE REVIEW

File: lib/spoom/git.rb [hunk 1]
```diff
@@ -14,11 +14,12 @@ def self.exec(command, *arg, path: '.')
       return "", "Error: `#{path}` is not a directory.", false unless File.directory?(path)
       opts = {}
       opts[:chdir] = path
-      _, o, e, s = Open3.popen3(*T.unsafe([command, *T.unsafe(arg), opts]))
+      i, o, e, s = Open3.popen3(*T.unsafe([command, *T.unsafe(arg), opts]))
       out = o.read.to_s
       o.close
       err = e.read.to_s
       e.close
+      i.close
```
Comment (vinistock) [inline lib/spoom/git.rb:22]: This was breaking locally with an error related to having too many files open from `Open3`.

File: test/spoom/cli/lsp_test.rb [hunk 2]
```diff
@@ -44,7 +44,7 @@ def foo(a)
         assert_equal(<<~MSG, err)
           Error: Sorbet returned typechecking errors for `/errors.rb`
             8:0-8:11: Not enough arguments provided for method `Foo#foo`. Expected: `1`, got: `0` (7004)
-            4:2-5:5: Returning value that does not conform to method result type (7005)
+            4:2-5:5: Expected `String` but found `NilClass` for method result type (7005)
```
Comment (vinistock) [inline test/spoom/cli/lsp_test.rb:47]: This changed after the Sorbet upgrade.

File: lib/spoom/git.rb (hunk 1 above)
Comment (Morriar) [reply lib/spoom/git.rb:22]: Good catch ❤️

**Review by Morriar — APPROVED** [review]

**Comment by Morriar** [comment]
Looks like the Sorbet bump also created errors around blocks: https://github.com/Shopify/spoom/pull/77/checks?check_run_id=2125606434 we might have to update those definitions upstream :/

**Review by RyanBrushett — APPROVED** [review]
Your CI failures are related to upstream RBI files defining a method that is missing an explicit block param. Happy to 🍐 to fix those if you're interested!

**Comment by vinistock** [comment]
Fixed the types for `with_clean_env` and some other related methods in https://github.com/sorbet/sorbet/pull/4092. Now waiting for the release to update this PR.

**Pull request merged**